### DISCUSSION TITLE
To refactor `BodyPartParser` of `Multipart` in order to avoid StackOverflowError easily

### DISCUSSION THREAD

**Discussion by yousuketto** [playframework/playframework#11347] [General]
The current code uses recaursive call so StackOverflowError is occurred. We need to estimate `maxMemoryBuffer` size in order to avoid the error.
https://github.com/playframework/playframework/issues/10590

I guess refactoring the code to use tailrec or loop would avoid the error without estimating the size of `maxMemoryBuffer`.
I think it can be done with a change like this.
https://github.com/yousuketto/playframework/commit/db119773acf92c6c211f5dac9c745f736f8b0973

**Comment by mkurz** [comment]
@yousuketto Feel free to submit a pull request with your suggestion, so we will see if tests pass and we can discuss further. Thanks!

**Reply by yousuketto** [reply]
I've created a pull request. Thank you.
https://github.com/playframework/playframework/pull/11360
